packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,295 @@
+import { Inject, Injectable, Optional } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { SendTransactionOptions, WalletNotConnectedError, WalletNotReadyError } from '@solana/wallet-adapter-base';
+import { WalletName } from '@solana/wallet-adapter-wallets';
+import { Connection, Transaction } from '@solana/web3.js';
+import { asyncScheduler, combineLatest, defer, from, Observable, of, throwError } from 'rxjs';
+import { catchError, concatMap, filter, first, map, observeOn, switchMap, tap, withLatestFrom } from 'rxjs/operators';
+
+import { fromAdapterEvent, isNotNull } from '../operators';
+import { WalletNotSelectedError } from './wallet.errors';
+import { WALLET_CONFIG } from './wallet.tokens';
+import { WalletConfig, WalletState } from './wallet.types';
+
+export const WALLET_DEFAULT_CONFIG: WalletConfig = {
+    wallets: [],
+    autoConnect: false,
+    localStorageKey: 'walletName',
+    onError: (error: unknown) => console.error(error),
+};
+
+@Injectable()
+export class WalletStore extends ComponentStore<WalletState> {
+    private readonly _autoConnect = this._config?.autoConnect || false;
+    private readonly _localStorageKey = this._config?.localStorageKey || 'walletName';
+    private logError = this._config?.onError || ((error: unknown) => console.error(error));
+    readonly wallets$ = this.select((state) => state.wallets);
+    readonly selectedWallet$ = this.select((state) => state.selectedWallet);
+    readonly connected$ = this.select((state) => state.connected);
+    readonly connecting$ = this.select((state) => state.connecting);
+    readonly disconnecting$ = this.select((state) => state.disconnecting);
+    readonly wallet$ = this.select((state) => state.wallet);
+    readonly adapter$ = this.select((state) => state.adapter);
+    readonly publicKey$ = this.select((state) => state.publicKey);
+    readonly ready$ = this.select((state) => state.ready);
+    readonly anchorWallet$ = this.select(
```

Comment:
How does a dapp reference this?

Reply:
Angular uses DI for accessing singletons, which is the case of the WalletStore. Consumers can do:

```typescript
class WildComponent {
    constructor(private walletStore: WalletStore) {}
}
``` 

With that in place, they have access to the walletStore singleton instance and they can do something like `this.walletStore.anchorWallet$` to access the observable. What I do in my dapp is:

```typescript
const anchorProvider$ = combineLatest([
  this.connectionStore.connection$,
  this.walletStore.anchorWallet$
]).pipe(
  map(
    ([connection, anchorWallet]) => 
      anchorWallet ? 
        new Provider(connection, anchorWallet, Provider.defaultOptions()) : 
        undefined
))
```

A responsible developer would also add an error notification of some kind which would teardown the observable and wouldn't continue.

Reply:
Got it, makes sense. Thanks.